### PR TITLE
Refactor/prevent duplicate favorites

### PR DESCRIPTION
**Pull request by shayan-golafshani**
## Description

In this PR, I added logic to prevent users from being able to add the same poem into their liked poems multiple times!
Then I added a message to inform the users that they've in fact, already liked the poem in their favorites, so the user isn't confused. Afterwards, I refactored that duplication prevention logic to cut down on lines, and verbosity. Lastly, I added a label for the select statement and a default value, guiding the user to pick an author.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor

## Motivation and Context
This PR, provides for a cleaner, more robust experience for the user that is bug free.

## Concerns
- No bugs!
-  Maybe I can change the CSS for the poems that are already like to have a heart with a green background instead of red like normal.

### CODE REVIEW

File: src/Components/Poems/Poems.js [hunk 1]
```diff
@@ -8,37 +8,28 @@ function Poem() {
...
-    //matchedPoem = matchedPoem[0]
-
-    //console.log("This is the matched poem", matchedPoem);
-    //let stringifiedPoem = JSON.stringify(matchedPoem[0])
-
     if(localData) {
-      //localData = JSON.parse(localData)
-      //console.log("LOCAL DATA inside Assignment IF", localData)
+      localData = JSON.parse(localData)
+      let found = localData.find(poem => poem.title === matchedPoem[0].title)
```
Comment (demaceo) [inline src/Components/Poems/Poems.js:21]: When searching for a poem & then favoriting songs, this error message appears.
Screen Shot 2021-08-08 at 3 31 42 PM

**Review by demaceo — APPROVED** [review]
Nice- really like this idea!

File: src/Components/Poems/Poems.js (hunk 1 above)
Comment (demaceo) [reply src/Components/Poems/Poems.js:21]: additionally (after getting results from one search result & then searching for another), the previous results do not get removed & the initially searched name is changed to the newly searched name in each of the poem cards

Comment (shayan-golafshani) [reply src/Components/Poems/Poems.js:21]: Thank you for breaking my app! I will look into the cause of it. And try async & await calls

Comment (demaceo) [reply src/Components/Poems/Poems.js:21]: anytime brochacho 😄

**Pull request merged**